### PR TITLE
e2e: make ForEach fail if filter is empty, fix no-op tests

### LINKED ISSUE #25426: Kubectl client Kubectl logs should be able to retrieve and filter logs [Conformance] failing on all 1.2 branch jobs
Example from [kubernetes-e2e-gce-release-1.2/2834](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-release-1.2/2834) ([jenkins page](http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-release-1.2/2834/consoleFull))

```
19:04:56 STEP: Destroying namespace "e2e-tests-kubectl-6sila" for this suite.
19:04:56 
19:04:56 
19:04:56 • Failure [59.548 seconds]
19:04:56 Kubectl client
19:04:56 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:1164
19:04:56   Kubectl logs
19:04:56   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:857
19:04:56     should be able to retrieve and filter logs [Conformance] [It]
19:04:56     /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:856
19:04:56 
19:04:56     Expected
19:04:56         <int>: 0
19:04:56     to equal
19:04:56         <int>: 1
19:04:56 
19:04:56     /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:831
```

It looks like the test started failing consistently yesterday at 7pm. cc @k8s-oncall

### PR DESCRIPTION
**Pull request by j3ffml**
Only review second commit (first is #25519).

### CODE REVIEW

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this issue: #23822

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -892,6 +892,14 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 		It("should be able to retrieve and filter logs [Conformance]", func() {
 			framework.SkipUnlessServerVersionGTE(extendedPodLogFilterVersion, c)
 
+			// Split("something\n", "\n" returns ["something", ""], so
```
Comment (spxtr) [inline test/e2e/kubectl.go]: nit: you're missing the closing `)` which makes this confusing to read.

**Comment by spxtr** [comment]
LG other than nit

**Comment by jlowdermilk** [comment]
Comments addressed.

**Comment by jlowdermilk** [comment]
Manually merging to fix tests.

**Pull request merged**